Controller/SettingsController.php

```diff
@@ -17,22 +15,22 @@ class SettingsController extends Controller {
 
 	public function modifyAction(Request $request) {
 		$em = $this->getDoctrine()->getManager();
-		$repo = $em->getRepository(get_class(new Setting()));
+		$repo = $em->getRepository('Craue\ConfigBundle\Entity\Setting');
 		$allStoredSettings = $repo->findAll();
 
 		$formData = array(
 			'settings' => $allStoredSettings,
 		);
 
-		$form = Kernel::VERSION_ID < 20800
-			? $this->createForm(new LegacyModifySettingsForm(), $formData)
-			: $this->get('form.factory')->createNamed('craue_config_modifySettings', 'Craue\ConfigBundle\Form\ModifySettingsForm', $formData);
+		$form = method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')
+			? $this->get('form.factory')->createNamed('craue_config_modifySettings', 'Craue\ConfigBundle\Form\ModifySettingsForm', $formData)
```

Comment:
if you define ``getBlockPrefix`` properly (and you should, otherwise you break BC for form theming), you would not need to use ``createNamed``, and would be able to use the shortcut here too